### ISSUE TITLE
Numpy float to iris Cell comparisons throwing error

### ISSUE THREAD

**Issue by Jozhogg** [SciTools/iris#1766]
```
import iris
import numpy as np

x = np.float32(1)
cell = iris.coords.Cell(3, [2,4])

# Returns false as expected
print(cell < x)

# Throws exception
print(x > cell)

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: setting an array element with a sequence.
```

I think the problem here is being caused by python calling `__ lt __` on the `np.float32` object rather than on the `iris.coords.Cell` object. Normally (e.g. for a python float) `__lt__` is not implemented for an `iris.coords.Cell` and so python defers to the operator definition of the Cell object. Unfortunately, numpy supports comparison between numpy floats and `collections.NamedTuple` which `iris.coords.Cell` is a subclass of.

This came up in a support ticket where somebody was deducing bounds for a constraint from the means of two numpy arrays (`iris.Constraint(time=lambda cell: x < cell < y)`). I wasn't able to find a proper fix for the issue.

**Comment by pelson** [comment]
May be related to the array priority. A similar fix was needed in https://github.com/SciTools/biggus/pull/176.

**Comment by pelson** [comment]
Fixed 4 days ago in #1947.
